packages/editor/src/extensions/audio/component.tsx

```diff
@@ -0,0 +1,158 @@
+
+  useEffect(() => {
+    if (editor.storage?.getAttachmentData && hash) {
+      editor.storage
+        .getAttachmentData({
+          type: "file",
+          hash
+        })
+        .then((data: string | undefined) => {
+          if (data) {
+            const url = toBlobURL(data, "other", mime, hash);
+            if (url) {
+              setAudioSrc(url);
+            }
+          }
+        })
+        .catch(console.error);
+    }
+  }, [editor.storage, hash, mime]);
+
+  useEffect(() => {
+    return () => {
+      if (audioSrc && hash) {
+        revokeBloburl(hash);
+      }
+    };
+  }, [audioSrc, hash]);
+
+  return (
+    <Box
+      ref={elementRef}
+      contentEditable={false}
+      draggable="false"
+      sx={{
+        display: "flex",
+        flexDirection: "column",
+        position: "relative",
+        userSelect: "none",
+        backgroundColor: "var(--background-secondary)",
+        p: 2,
+        borderRadius: "default",
+        border: "1px solid var(--border)",
+        borderColor: selected ? "accent" : "border",
+        width: "100%",
+        maxWidth: 500,
+        ":hover": {
+          backgroundColor: "var(--hover)"
+        }
+      }}
+      onDragStart={() => setIsDragging(true)}
+      onDragEnd={() => setIsDragging(false)}
+      data-drag-handle
+    >
+      <Box
+        sx={{
+          display: "flex",
+          alignItems: "center",
+          mb: audioSrc ? 2 : 0
+        }}
+      >
+        <Icon path={Icons.attachment} size={16} />
+        <Text
+          as="span"
+          sx={{
+            ml: "small",
+            fontSize: "body",
+            whiteSpace: "nowrap",
+            textOverflow: "ellipsis",
+            overflow: "hidden",
+            flex: 1
+          }}
+        >
+          {filename}
+        </Text>
+        <Text
+          as="span"
+          sx={{
+            ml: 1,
+            fontSize: "0.65rem",
+            color: "var(--paragraph-secondary)",
+            flexShrink: 0
+          }}
+        >
+          {progress ? `${progress}%` : formatBytes(size)}
+        </Text>
+      </Box>
+      {audioSrc && (
+        <Box
+          sx={{
+            width: "100%",
+            "& audio": {
+              width: "100%",
+              height: 40
+            }
+          }}
+        >
+          <audio controls preload="metadata" src={audioSrc} />
+        </Box>
+      )}
+      <DesktopOnly>
+        {selected && !isDragging && (
+          <ToolbarGroup
+            editor={editor}
+            groupId="audioTools"
+            tools={
+              editor.isEditable
+                ? ["removeAttachment", "downloadAttachment"]
+                : ["downloadAttachment"]
+            }
+            sx={{
+              boxShadow: "menu",
+              borderRadius: "default",
+              backgroundColor: "var(--background)",
+              position: "absolute",
+              top: -35
+            }}
+          />
```

Comment:
nit: instead of top left, can we have the toolbar at the top right corner (similar to the image toolbar)

<img width="754" height="254" alt="Image" src="https://github.com/user-attachments/assets/7bb02f0f-5663-4eac-a1bf-90a375d88c70" />